python_modules/dagster/dagster/_core/definitions/step_launcher.py

```diff
@@ -56,6 +57,14 @@ def __new__(
         )
 
 
+_step_launcher_deprecation = deprecated(
```

Comment:
given the taxonomy, i believe we'd want to emit `FutureWarning` to communicate supersession rather than `DeprecationWarning`

Reply:
Made a downstack PR here: https://github.com/dagster-io/dagster/pull/25363